### PR TITLE
Turning on and switching internals now takes time.

### PR DESCRIPTION
**Pull request by TheBoondock**
## About The Pull Request
This pr makes turning and switching internals take 4 seconds to do.

## Why It's Good For The Game
Right now most of atmos hazards can be completely nulify and ignored instantly with a click of a button. Given that atmos can be slow at time it doesnt take a quick reaction time to turn on internal at the sight of atmos hazard and ignore it completely. This pr aims to make atmos hazard more dangerous by making internals takes time to switch on.

## Changelog

:cl:
balance: turning on and switching internals now takes 4 seconds
/:cl:

### CODE REVIEW

**Comment by castawaynont** [comment]
>sleeping gas flood
>don't notice in time, fall asleep
you are literally better off alt tabbing

**Comment by red-creature** [comment]
i sure do love getting permanent sleep until i die because i didn't notice the sleep gas in the air. when i wake up i can't turn on my internals before i just pass out again. this is fun game design.

File: code/game/objects/items/tanks/tanks.dm [hunk 1]
```diff
@@ -69,6 +69,9 @@
 
 		if(H.internal)
 			to_chat(H, span_notice("You switch your internals to [src]."))
+		to_chat(H, span_notice("You fumble with the connector."))
+		if(!do_after(H, 4 SECONDS, H.internal, IGNORE_USER_LOC_CHANGE))
+			return TRUE
 		else
 			to_chat(H, span_notice("You open [src] valve."))
```
Comment (ATH1909) [inline code/game/objects/items/tanks/tanks.dm:76]:
```suggestion
		to_chat(H, span_notice("You fumble with the connector."))
		if(!do_after(H, 4 SECONDS, H.internal, IGNORE_USER_LOC_CHANGE))
			return TRUE
		if(H.internal)
			to_chat(H, span_notice("You switch your internals to [src]."))
		else
			to_chat(H, span_notice("You open [src] valve."))
```

**Comment by SuperNovaa41** [comment]
> Right now most of atmos hazards can be completely nulify and ignored instantly with a click of a button. 

i mean, theres still the whole process of equipping the mask and the o2 tank. n2o floods already prove that a lot of people are incapable of doing that, this is just going to permasleep people in that case

**Comment by ATH1909** [comment]
please god no n2o will be a stunlock

also canister floods will be really degenerate with this in place

**Comment by iprice** [comment]
Smoke ingestion?

Some of these things can kill super fast.  CO2 particularly is a race to handle before you die.  Not everyone has a tank in the pocket all the time.  Or maybe we'll have to now.  Which would be sad.  At which point might as well just leave the things on all the time.

Idk that this is a good change, and probably some gas effects may need to be made somewhat slower to keep things reasonably reactable.

The only change any player will experience is frustration at watching the progress bar get 90% of the way there then they fall over.  The only 'gain' is that someone who likely doesn't even directly experience or participate in the victim's death gets +1 on their kill count.

**Comment by SkeletalElite** [comment]
If you're gonna add this there has to be a hold breath button that will stop you from breathing (causing you take o2 damage until you pass out or start breathing again of course) otherwise some gas mixtures will just straight up kill you which encourages people to game even harder by always have their internals on and just grabbing a new from a locker every few minutes or refilling at an o2 tank/grabbing one of the bigger yellow tanks from engi.

**Comment by Iamgoofball** [comment]
this is a bad design idea

**Comment by Kylerace** [comment]
no this sucks

**Comment by LemonInTheDark** [comment]
For the record, I disagree with the goal here. If we wanted to make gas more spooky we could, I don't think we should. That's what you needed to argue for.

Also n2o exists

**Pull request closed without merging**